internal/runners/deploy/uninstall/uninstall.go

```diff
@@ -66,19 +74,33 @@ func (u *Uninstall) Run(params *Params) error {
 	}
 
 	logging.Debug("Attempting to uninstall deployment at %s", path)
-	if !store.New(path).HasMarker() {
+	store := store.New(path)
+	if !store.HasMarker() {
 		return errs.AddTips(
 			locale.NewError("err_deploy_uninstall_not_deployed", "There is no deployed runtime at '{{.V0}}' to uninstall.", path),
 			locale.Tl("err_deploy_uninstall_not_deployed_tip", "Either change the current directory to a deployment or supply '--path <path>' arguments."))
 	}
 
+	var namespace string
+	var err error
+	namespace, err = store.Namespace()
+	if err != nil {
+		return locale.WrapError(err, "err_deploy_uninstall_namespace", "Unable to read namespace from marker file.")
+	}
+
+	var commitID string
+	commitID, err = store.CommitID()
+	if err != nil {
+		return locale.WrapError(err, "err_deploy_uninstall_commit_id", "Unable to read commit ID from marker file.")
+	}
```

Comment:
Analytics should never break a command. If we can't collect these then we can't report them, but it should not affect the action the user was taking.

Reply:
Also can we put this and the analytics call itself into a helper function? This is a lot of analytics code obfuscating the rest of the logic.

Reply:
I can move the sourcing of the commitID and namespace to a helper function but I assumed we wanted to fire the analytics event after successfully uninstalling the deployment, which would mean that the file we're using would be gone.

Reply:
You could read the file when the store is initialized rather than when the function is called. Or use 2 helper functions instead of 1. Main point here was getting the analytics code to be minimal inside the run function.

Reply:
I've moved the sourcing of the info to a helper function which is called before we delete the marker file. The analytics call is still in the runner as the helper would simply be wrapping that.